usaspending_api/common/filters/time_period/time_period.py

```diff
@@ -50,11 +51,31 @@ def gte_date_type(self):
         pass
 
     @abstractmethod
-    def lte_date_type(self):
+    def lte_date_type(self) -> str:
         """
         Returns:
             A date type that indicates the name of a column in which
-            stores the date values that end date must be less than.
+            stores the date values that end date must be less than or equal to.
             Can return None when lte_date_type doesn't exist.
         """
         pass
```

Comment:
If this is returning a column name, `lte_date_type` and `gte_date_type` may not be the best choices for this function. `Type` implies a class type (like `datetime` or `timespan` etc.). 

Can this just say `lte_date_field` and `gte_date_field` or replace field with "col" or "column" or "field_name" or "column_name" etc.

Reply:
Yep, I'll make that change

Reply:
Wait, no date type is exactly what we call it in an incoming request to our api. That's why it says date type because the field in the request is `date_type`

Reply:
But in moving these common filters outside the api realm, it should have no relation to or expectations of knowledge what the API calls things or their request/response payloads. If this is in fact very API specific, it should be moved under API namespaces. 
The way these filters are used is to apply them directly to domain/ORM objects when filtering data, so it shouldn't be coupled to the API request/response life cycle.

Reply:
These classes are for our API's filter dictionary.

Reply:
So we need to move them, they are not intended for any other filters.

Reply:
I don't understand how these classes are outside the api realm. They are under `usaspending_api`, how is that not considered the api realm?